quiche/src/packet.rs

```diff
@@ -555,20 +555,12 @@ impl<'a> std::fmt::Debug for Header<'a> {
     }
 }
 
-pub fn pkt_num_len(pn: u64) -> Result<usize> {
-    let len = if pn < u64::from(u8::MAX) {
-        1
-    } else if pn < u64::from(u16::MAX) {
-        2
-    } else if pn < 16_777_215u64 {
-        3
-    } else if pn < u64::from(u32::MAX) {
-        4
-    } else {
-        return Err(Error::InvalidPacket);
-    };
-
-    Ok(len)
+pub fn pkt_num_len(pn: u64, largest_acked: u64) -> Result<usize> {
+    let num_unacked: u64 = pn.checked_sub(largest_acked).map_or(0, |v| v) + 1;
+    // computes ceil of num_unacked.log2()
+    let min_bits = u64::BITS - num_unacked.leading_zeros();
+    // get the num len in bytes
+    Ok((min_bits as f32 / 8.).ceil() as usize)
```

Comment:
`Ok((num_bits + 7) / 8 as usize)` should be faster.

Reply:
Looks right! Good catch. The Result is not useful anymore either, and I removed it.